File: Easy/94. Binary Tree Inorder Traversal/solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    
    public void InorderTraversalHelper(TreeNode root, ref List<int> result){
        if (root == null)   return;
        InorderTraversalHelper(root.left, ref result);
        //Console.Write(root.val + " ");
        result.Add(root.val);
        InorderTraversalHelper(root.right, ref result);
    }
    
    public IList<int> InorderTraversal(TreeNode root) {
        List<int> result = new List<int> ();
        
        InorderTraversalHelper(root, ref result);
        
        return result;
    }
}

File: Easy/94. Binary Tree Inorder Traversal/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        
        stack<TreeNode*> myStack;
        TreeNode* curNode = root;
        
        vector<int> result;
        
        while (curNode || !myStack.empty()){
            if (curNode){
                myStack.push(curNode);
                curNode = curNode -> left;
            }
            else{
                curNode = myStack.top();
                myStack.pop();
                result.push_back(curNode -> val);
                curNode = curNode -> right;
            }
        }
        
        return result;
    }
};